### ISSUE TITLE
Glitch in mafs/vector-field

### ISSUE THREAD

**Issue by alexgian** [mentat-collective/emmy-viewers#78]
This may be a Mafs issue, but I thought I'd report it here anyway.

The following code uses `mafs/vector-field` to create a makeshift slope-field display.
```
(defn xdash [x y] (+ y))
(defn ydash [x y] (- x))

(defn fx [[x y]]
  (let [x1 (xdash x y)
        y1 (ydash x y)
        norm (sqrt (+ (square x1) (square y1)))
        xout (/ x1 norm)
        yout (/ y1 norm)]
    [xout yout]))

(mafs/mafs {:height 400 :zoom {:min 0.50 :max 2.0}}
           (mafs/cartesian {:subdivisions 4})
           (mafs/vector-field {:step       0.25001
                               :xy         fx
                               :xy-opacity (constantly 0.5)
                               }))
```
try it with values like a polynomial:
```
(defn xdash [x y] (+ 1))
(defn ydash [x y] (+ (* 3 x) 2)
```
 exponential:
```
(defn xdash [x y] (+ 1))
(defn ydash [x y] (+ y))
```
trig waves:
```
(defn xdash [x y] (cos y))
(defn ydash [x y] (sin x))
```
and it works fine.  However, with the given values (system is equivalent to f''=-f) the first quadrant does not display.  This seems to happen also whenever the absolute values of xdash and ydash are equal, assuming x=y.  E.g
```
(defn xdash [x y] (* 2 y))
(defn ydash [x y] (* -2 x))
```
will also do it.

If that's not strange enough, notice the peculiar `:step` key I use.  If I change that to 0.25 (or to some multiples of it, i.e. 0.125, 0.5, 1.0, 2) then practically *three* quadrants do not display!

**Comment by sritchie** [comment]
I filed a mafs issue here... https://github.com/stevenpetryk/mafs/issues/164

This is an issue with mafs, but when they fix it we can update the version and close this ticket out.

**Comment by sritchie** [comment]
Looks like it happens with simpler functions, like `identity`!

**Comment by sritchie** [comment]
This is fixed in mafs 0.18.8, as mentioned here: https://github.com/stevenpetryk/mafs/issues/164

To close this issue we'll need to upgrade mafs.cljs to 0.18.8 and pull in the new version here.

**Comment by sritchie** [comment]
emmy-viewers 0.3.2 closes this: https://clojars.org/org.mentat/emmy-viewers

Thanks for the report!
